flink-runtime/src/main/java/org/apache/flink/runtime/resourcemanager/slotmanager/FineGrainedSlotManager.java

```diff
@@ -739,9 +789,8 @@ private void releaseIdleTaskExecutorIfPossible(TaskManagerInfo taskManagerInfo)
 
     private void releaseIdleTaskExecutor(InstanceID timedOutTaskManagerId) {
         if (resourceAllocator.isSupported()) {
-            final FlinkExpectedException cause =
-                    new FlinkExpectedException("TaskManager exceeded the idle timeout.");
-            resourceAllocator.releaseResource(timedOutTaskManagerId, cause);
+            taskManagerTracker.addUnWantedTaskManager(timedOutTaskManagerId);
+            declareNeededResources();
```

Comment:
I think we still need the delay mechanism, to avoid too frequent declarations. E.g., we may have tens or even hundreds of idle taskmanagers being found in the same round of idleness check, which means this method `releaseIdleTaskExecutor` can be called hundreds of times within very short time, each triggers a processing of the full declarations. A very short delay (e.g., 50ms) would significantly help on this.